client.go

```diff
@@ -128,43 +128,36 @@ func (c *Client) InterpolatePath(path string, params ...string) (string, error)
 	return fmt.Sprintf(template, encodedParams...), nil
 }
 
-// HttpCaller is the generic http interface used by the Client
-type HttpCaller interface {
-	Call(method string, path string, genericParams GenericParams, v interface{}) error
+// HTTPCaller is the generic http interface used by the Client
+type HTTPCaller interface {
+	Call(ctx context.Context, method string, path string, body interface{}, queryParams QueryParams, requestOptions optionsApplier, v interface{}) error
```

Comment:
In `3.x` the `body` has to be a `GenericParams` to avoid a breaking change. In reality, the `GenericParams` in 3.x is a mechanism to convert to a `Params` (removed in 4.x) so that the `Params.Data` can be accessed. `Params.Data` is just an empty `interface{}` in 3.x